exchange/src/main/java/org/geneontology/gocam/exchange/BioPaxtoGO.java

```diff
@@ -1500,6 +1484,10 @@ private IRI getPhysicalEntityIRI(Entity entity) {
 			}else if(entityStrategy.equals(EntityStrategy.YeastCyc)) {
 				if(entity_id.startsWith("CHEBI")) {
 					IRI entity_class_iri = IRI.create("http://purl.obolibrary.org/obo/"+entity_id);
+					if(golego.chebi_roles.contains(entity_class_iri.toString())) {
+						// No roles of any kind can be treated as physical entities - fall back on 'chemical entity'
+						return IRI.create("http://purl.obolibrary.org/obo/CHEBI_24431");
```

Comment:
Would it be possible to add the role as well, related via 'has role'?

Reply:
Sorry, I just noticed this suggestion. I [asked](https://github.com/geneontology/pathways2GO/issues/139#issuecomment-1109106771) @thomaspd if we should add this.

Reply:
See https://github.com/geneontology/pathways2GO/issues/139#issuecomment-1118860983